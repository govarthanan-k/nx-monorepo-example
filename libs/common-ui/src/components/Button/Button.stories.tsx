import type { Meta, StoryObj } from "@storybook/react";
import { expect, userEvent, within } from "@storybook/test";
import { Button } from "./Button";

const meta: Meta<typeof Button> = {
  component: Button,
  title: "Components/Button",
};
export default meta;

type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    text: "Click me",
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole("button", { name: /click me/i });

    // Simple assertions
    await expect(button).toBeInTheDocument();
    await expect(button).toHaveTextContent("Click me");

    // Simple interaction
    await userEvent.click(button);

    console.log("✅ Primary button test passed");
  },
};

export const Secondary: Story = {
  args: {
    text: "Click me",
    variant: "red",
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole("button", { name: /click me/i });

    await expect(button).toBeInTheDocument();
    await expect(button).toHaveTextContent("Click me");
    await userEvent.click(button);

    console.log("✅ Secondary button test passed");
  },
};

// Simple story without complex interactions
export const InteractiveTest: Story = {
  args: {
    text: "Interactive Test",
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole("button");

    // Basic tests only
    await expect(button).toBeVisible();
    await userEvent.click(button);

    console.log("✅ Interactive test passed");
  },
};
