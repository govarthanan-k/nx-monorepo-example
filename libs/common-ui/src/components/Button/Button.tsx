import type { ButtonProps } from "./Button.types";

export const Button = (props: ButtonProps) => {
  const { text, variant = "blue" } = props;

  return <button style={{ background: variant }}>{text}</button>;
};
