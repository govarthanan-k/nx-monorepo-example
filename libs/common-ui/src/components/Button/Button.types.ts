export interface ButtonProps {
  text: string;
  variant?: "blue" | "red";
}
