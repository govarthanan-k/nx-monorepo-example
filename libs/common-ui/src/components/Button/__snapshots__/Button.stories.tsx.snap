// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Button InteractiveTest play-test: Components/Button-Interactive Test.html 1`] = `
<button style="background: blue;">
  Interactive Test
</button>
`;

exports[`Components/Button Primary play-test: Components/Button-Primary.html 1`] = `
<button style="background: blue;">
  Click me
</button>
`;

exports[`Components/Button Secondary play-test: Components/Button-Secondary.html 1`] = `
<button style="background: red;">
  Click me
</button>
`;
